### PR TITLE
Fix #28552 Add test for DataFrame min/max preserve timezone

### LINKED ISSUE #28552: DataFrame[list[Series[datetime64ns, tz]]] drops timezone information
#### Code Sample, a copy-pastable example if possible

```
s = pd.Series([pd.to_datetime('2018-10-08 13:36:45+00:00')])


Output:
0   2018-10-08 13:36:45+00:00
dtype: datetime64[ns, UTC]
```

```
pd.DataFrame([s]).min()


Output:
0   2018-10-08 13:36:45
dtype: datetime64[ns]
```

#### Problem description

When using `DataFrame().min()` method timezone information gets removed but it should remain untouched just like when using `Series().combine(...,min)`. Example:

```
s.combine(s, min)


Output:
0   2018-10-08 13:36:45+00:00
dtype: datetime64[ns, UTC]
```

#### Expected Output
```
0   2018-10-08 13:36:45+00:00
dtype: datetime64[ns, UTC]
```
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.6.7.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-64-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.0
numpy            : 1.17.0
pytz             : 2019.2
dateutil         : 2.8.0
pip              : 19.2.1
setuptools       : 40.6.3
Cython           : 0.27.3
pytest           : 3.0.7
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.2.6
html5lib         : None
pymysql          : None
psycopg2         : 2.7.3.2 (dt dec pq3 ext lo64)
jinja2           : 2.10
IPython          : 7.2.0
pandas_datareader: None
bs4              : 4.6.3
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.2.6
matplotlib       : 2.0.2
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : 0.11.0
pyarrow          : 0.13.0
pytables         : None
s3fs             : None
scipy            : 1.0.0
sqlalchemy       : 1.3.7
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by ghost**
- [x] closes #28552
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry _(not applicable - no functionality added)_

Here's a test discussed in #28552. `mean` and other aggregates except for `min` and `max` are not covered since they don't support datetimes - it's a nan being returned.

### CODE REVIEW

File: pandas/tests/frame/test_timezones.py [hunk 1]
```diff
@@ -166,3 +166,19 @@ def test_constructor_data_aware_dtype_naive(self, tz_aware_fixture):
         result = DataFrame({"d": [pd.Timestamp("2019", tz=tz)]}, dtype="datetime64[ns]")
         expected = DataFrame({"d": [pd.Timestamp("2019")]})
         tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "initial",
+        ["2018-10-08 13:36:45+00:00", "2018-10-08 13:36:45+03:00"],  # Non-UTC timezone
+    )
+    @pytest.mark.parametrize("method", [DataFrame.min, DataFrame.max])
```
Comment (mroeschke) [inline pandas/tests/frame/test_timezones.py:174]: Could you use the strings `['min', 'max']` here, and then below you can do `getattr(df, method)(axis=1)`

File: pandas/tests/frame/test_timezones.py [hunk 2]
```diff
@@ -166,3 +166,19 @@ def test_constructor_data_aware_dtype_naive(self, tz_aware_fixture):
...
+    def test_preserve_timezone(self, initial: str, method):
+        # GH 28552
+        # Given
+        initial_dt = pd.to_datetime(initial)
+        series = pd.Series([initial_dt])
+        df = DataFrame([series])
+        # When
+        actual_df = method(df, axis=1)
+        # Then
+        assert actual_df[0].tz == initial_dt.tz
```
Comment (mroeschke) [inline pandas/tests/frame/test_timezones.py:184]: Could you rename the results `result` and `expected` and compare 2 DataFrame or Series objects and use `tm.assert_equal(result, expected)`

Comment (dair-targ) [reply pandas/tests/frame/test_timezones.py]: Addressed. Please note that `min` and `max` methods of the `DataFrame` both return `Series`, so I choose to use the `tm.assert_series_equal` instead in the same manner as it is being used in `test_frame_no_datetime64_dtype`.

File: pandas/tests/frame/test_timezones.py (hunk 1 above)
Comment (dair-targ) [reply pandas/tests/frame/test_timezones.py]: Addressed.

**Comment by dair-targ** [comment]
@mroeschke thanks for review! I hope I've addressed all the issues - would you mind to review it again?

**Review by mroeschke — APPROVED** [review]
LGTM

File: pandas/tests/frame/test_timezones.py [hunk 3]
```diff
@@ -166,3 +166,19 @@ def test_constructor_data_aware_dtype_naive(self, tz_aware_fixture):
...
+
+    @pytest.mark.parametrize(
+        "initial",
+        ["2018-10-08 13:36:45+00:00", "2018-10-08 13:36:45+03:00"],  # Non-UTC timezone
+    )
+    @pytest.mark.parametrize("method", ["min", "max"])
+    def test_preserve_timezone(self, initial: str, method):
+        # GH 28552
+        # Given
+        initial_dt = pd.to_datetime(initial)
```
Comment (jreback) [inline pandas/tests/frame/test_timezones.py:178]: can you move to pandas/tests/frame/test_analytics.py (at the very end).

File: pandas/tests/frame/test_timezones.py [hunk 4]
```diff
@@ -166,3 +166,19 @@ def test_constructor_data_aware_dtype_naive(self, tz_aware_fixture):
...
+        expected = pd.Series([initial_dt])
+        df = DataFrame([expected])
+        # When
```
Comment (jreback) [inline pandas/tests/frame/test_timezones.py:181]: you don't need the comments here

File: pandas/tests/frame/test_timezones.py [hunk 5]
```diff
@@ -166,3 +166,19 @@ def test_constructor_data_aware_dtype_naive(self, tz_aware_fixture):
...
+        result: pd.Series = getattr(df, method)(axis=1)
```
Comment (jreback) [inline pandas/tests/frame/test_timezones.py:182]: the type annotation here is unecessary

File: pandas/tests/frame/test_timezones.py [hunk 6]
```diff
@@ -166,3 +166,19 @@ def test_constructor_data_aware_dtype_naive(self, tz_aware_fixture):
...
+    @pytest.mark.parametrize(
+        "initial",
+        ["2018-10-08 13:36:45+00:00", "2018-10-08 13:36:45+03:00"],  # Non-UTC timezone
+    )
+    @pytest.mark.parametrize("method", ["min", "max"])
+    def test_preserve_timezone(self, initial: str, method):
+        # GH 28552
+        # Given
+        initial_dt = pd.to_datetime(initial)
+        expected = pd.Series([initial_dt])
```
Comment (jreback) [inline pandas/tests/frame/test_timezones.py:179]: user Series (no pd)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_timezones.py (hunk 3 above)
Comment (dair-targ) [reply pandas/tests/frame/test_timezones.py:178]: Do you mean to have this test as a standalone function, or move it to be a method of `TestDataFrameReductions`?

File: pandas/tests/frame/test_timezones.py (hunk 6 above)
Comment (dair-targ) [reply pandas/tests/frame/test_timezones.py:179]: `pd.Series` occurs 39 times in `test_analytics.py` while just ` Series` occurs 24 times. Both seems to occur in relatively new commits. I'm ok to do the change, but do you have some fixed code style on that?

File: pandas/tests/frame/test_timezones.py (hunk 3 above)
Comment (jreback) [reply pandas/tests/frame/test_timezones.py:178]: i think as an additional test in TestDataFrameReductions is fine (we are going to split this out at some point anyhow to a new file)

File: pandas/tests/frame/test_timezones.py (hunk 6 above)
Comment (jreback) [reply pandas/tests/frame/test_timezones.py:179]: yeah we haven't settled on a singular strategy yet. so ok to leave

File: pandas/tests/frame/test_timezones.py (hunk 3 above)
Comment (dair-targ) [reply pandas/tests/frame/test_timezones.py]: Done.

File: pandas/tests/frame/test_timezones.py (hunk 5 above)
Comment (dair-targ) [reply pandas/tests/frame/test_timezones.py]: Removed.

File: pandas/tests/frame/test_timezones.py (hunk 4 above)
Comment (dair-targ) [reply pandas/tests/frame/test_timezones.py]: Removed.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @dair-targ

**Pull request merged**